### PR TITLE
(robot_plugin) load all visuals

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
This fixes the missing shapes of the robot in the web interface.

### CODE REVIEW

File: plugins/robot_plugin.cpp [hunk 1]
```diff
@@ -154,12 +148,36 @@ geo::ShapePtr linkToShape(const urdf::LinkSharedPtr& link)
...
+        return shape;
+
+    for (urdf::VisualSharedPtr& vis : link->visual_array)
     {
-        geo::ShapePtr shape_tr(new geo::Shape);
-        shape_tr->setMesh(shape->getMesh().getTransformed(offset));
-        shape = shape_tr;
+        const urdf::GeometrySharedPtr& geom = vis->geometry;
+        if (!geom)
+            continue;
```
Comment (PetervDooren) [inline plugins/robot_plugin.cpp:167]: Maybe add a warning that there is a model error

File: plugins/robot_plugin.cpp [hunk 2]
```diff
@@ -154,12 +148,36 @@ geo::ShapePtr linkToShape(const urdf::LinkSharedPtr& link)
...
+
+        geo::Pose3D offset;
+        const urdf::Pose& o = vis->origin;
+        offset.t = geo::Vector3(o.position.x, o.position.y, o.position.z);
+        offset.R.setRotation(geo::Quaternion(o.rotation.x, o.rotation.y, o.rotation.z, o.rotation.w));
+
+        geo::ShapePtr subshape = URDFGeometryToShape(geom);
+        if (!subshape)
+            continue;
```
Comment (PetervDooren) [inline plugins/robot_plugin.cpp:176]: add logging

**Comment by PetervDooren** [comment]
For better logging, dont just say: mesh could not be made. Print what geom.get() gives as this is the first thing someone wants to know

**Comment by MatthijsBurgh** [comment]
> For better logging, dont just say: mesh could not be made. Print what geom.get() gives as this is the first thing someone wants to know

I have added the filename which failed to be loaded.

All the `geom.get()` is a pointer. So in case this is false. It is a `nullptr`. So there is no info to be shown. The mesh can still fail. I think the logging of the filename is sufficient now.
The other shapes don't fail.

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**